src/msg.rs

```diff
@@ -6,27 +6,45 @@ use serde::{Deserialize, Serialize};
 pub struct InstantiateMsg {
     pub vault_denom: String,
     pub amount_to_stake_per_validator: Uint128,
-    pub max_records_to_update_per_run: u32,
+    pub batch_size: u32,
 }
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
 pub enum ExecuteMsg {
     RecordMetrics { timestamp: u64 }, // 12:00AM
     AddValidator { addr: Addr },
-    UpdateRecordsToUpdatePerRun { no: u32 },
+    UpdateConfig { batch_size: u32 },
 }
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
 pub enum QueryMsg {
-    GetHistoryByTime {
+    GetAllTimestamps {},
+    GetAllValidatorMetrics {
+        addr: Addr,
+    },
+    GetValidatorMetricsBtwTimestamps {
+        addr: Addr,
+        timestamp1: u64,
+        timestamp2: u64,
+    },
+    GetValidatorMetricsByTimestamp {
+        timestamp: u64,
+        addr: Addr,
+    },
+    GetValidatorsMetricsByTimestamp {
         timestamp: u64,
+        from: u32,
+        to: u32,
     },
     GetState {},
-    GetAllAprsByInteral {
+    GetConfig {},
+    GetAllAprsByInterval {
         timestamp1: u64,
         timestamp2: u64,
+        from: u32,
```

Comment:
If you had made from, to, and timestamp all u64 it would have been easier for you.